### PR TITLE
acme: improve

### PR DESCRIPTION
**Pull request by hgl**
Maintainer: @tohojo 
Compile tested: x64 master
Run tested: x64 master

Description:

Please check the commit messages for details. Two changes:

1. Don't let clients deal with deprecated values.
2. Standardize key type values.

### CODE REVIEW

**Comment by stokito** [comment]
I see. And the keylength was renamed to keytype is now?
How the param is called in the acmesh itself?

I just trying to figure out what was changed  and where it's going

File: net/acme-common/files/acme.init [hunk 1]
```diff
@@ -39,8 +39,17 @@ load_options() {
...
-	export keylength
+	config_get keylength "$section" keylength
+	if [ "$keylength" ]; then
+		log warn "Option \"keylength\" is deprecated, please use key_type (e.g., ec256, rsa2048) instead."
+		case $keylength in
+		ec-*) key_type=${keylength/-/} ;;
+		*) key_type=rsa$keylength ;;
+		esac
+	else
+		config_get key_type "$section" key_type ec256
```
Comment (tohojo) [inline net/acme-common/files/acme.init:50]: Hmm, if someone actually sets the new value, the old one probably shouldn't override it, should it?

Comment (hgl) [reply net/acme-common/files/acme.init:50]: I assumed the convention was that if the old value exists, a user probably hasn’t completely migrated yet.

If you want to change that to “as soon so the new value comes into being“ I’m fine with that too.

Let me know.

**Comment by hgl** [comment]
> Oh, I see. And the keylength was renamed to keytype is How the param is called in the acmesh itself?
> 
> I just trying to figure out what was changed and where it's going

Old value is how acmesh calls it.

File: net/acme-common/files/acme.init (hunk 1 above)
Comment (tohojo) [reply net/acme-common/files/acme.init:50]: Hmm, I guess we also do the "old name first, then new" for the `staging` option, so let's just keep it here as well to stay consistent...

**Comment by tohojo** [comment]
Do you actually have another client in mind that will get confused by the current keylength values? Doesn't really seem like one is inherently superior to the other?

Also, luci-app-acme still uses the old config values, not just for this but also some of the other deprecated config names (use_staging and webroot). We should update that as well...

**Comment by stokito** [comment]
@tohojo if you don't have a time I can update the luci-app on next week.

**Comment by tohojo** [comment]
@stokito that would be awesome, thanks! :)

**Comment by hgl** [comment]
> Do you actually have another client in mind that will get confused by the current keylength values?

I'm writing a client using lego, but it's a WIP. Strictly speaking, it won't get confused, since we will write value converter, if the client's supported values are different. I think the point is, if the values have to be converted (which are very likely for non-acmesh clients), let's convert from ones that are more intuitive. Converting from pure numbers to rsa is not a good design IMHO.

> if you don't have a time I can update the luci-app on next week.

Thanks for the help @stokito.

**Comment by tohojo** [comment]
Ping @stokito on the luci update? :)

**Pull request merged**